webrecorder/browsertrix
# Stop, cancel, and download QA run

Partially addresses https://github.com/webrecorder/browsertrix-cloud/issues/1651

### Changes

- Disables review actions if analysis is running
- Enables downloading run
- Enables stopping and canceling runs
- Minor refactor to create and implement `btrix-menu-item-link` for download menu option
- Hides canceled QA runs from dropdown

### Manual testing

1. Log in as crawler
2. Go to Archived Items -> click a crawl
3. Click "[re]Run Analysis". Verify stop and cancel buttons are shown
4. Click "Stop". Verify confirmation dialog is shown
5. Click "Keep running". Verify analysis continues to run
6. Click "Stop" again and then "Stop Analysis". Leave tab and revisit tab. Verify analysis is stopped or stopping (see follow-ups for note on polling updates)
7. Repeat for "Cancel"
8. Click "Analysis Runs"
9. Click "..." action overflow menu for a canceled run. Verify download option is not visible
10. Click "..." for a completed run. Verify download option is enabled after a short delay
11. Click "Download". Verify QA run is downloaded

### Follow-ups
Following isn't handled in this PR
- Polling for currently running QA run (should be in `frontend-qa-overview-page`)
- Download multiple resource files. Changes in this PR only downloads the first file
